src/sentry/tagstore/multi/backend.py

```diff
@@ -236,3 +236,6 @@ def get_group_tag_value_qs(self, *args, **kwargs):
 
     def update_group_for_events(self, *args, **kwargs):
         return self._call_all_backends('update_group_for_events', *args, **kwargs)
+
+    def get_groups_times_seen_by_environment_name(self, *args, **kwargs):
```

Comment:
`abc.ABCMeta` is far from perfect but might be worth considering adding to `TagStorage` to avoid situations like this

Reply:
I glanced at the docs but I don't follow what it would do here?

Reply:
It'd prevent the class from being instantiated if an abstract method hasn't been overridden with a non-abstract implementation. This got caught because of incidental test coverage — if it was defined as an abstract method, this would have been caught on instantiation of the backend, regardless of if the method is ever called or not during the test suite.

Reply:
Ohhh, cool. Yes, we need that.